test/e2e/run_test.go

```diff
@@ -1270,8 +1270,12 @@ VOLUME %s`, ALPINE, volPath, volPath)
 		session.WaitWithDefaultTimeout()
 		Expect(session).Should(Exit(0))
 		Expect(session.OutputToString()).To(ContainSubstring("shared"))
-		// make sure it's only shared (and not 'shared,slave')
-		Expect(session.OutputToString()).To(Not(ContainSubstring("shared,")))
+
+		// this works only if the user owns the outer mount namespace.
+		if !isRootless() {
+			// make sure it's only shared (and not 'shared,slave')
+			Expect(session.OutputToString()).To(Not(ContainSubstring("shared,")))
```

Comment:
That seems like an extremely fragile test. Shouldn't this check for `slave` then and not `shared,`??

Reply:
the expected output is `"shared"`, not anything like "`shared,$SOMETHING_ELSE`.  I've left the original comment that mentions `slave` but it could also be `private`, but in either case it is wrong.  I think the test is correct, it checks for `shared` but not `shared,`.

How would you change it?

Reply:
Well first I do not fully understand what the test is after.

However the more precise way to check things is to check for Equal() and not substrings.
So assuming we only care about the mountpoint `/myvol2` the proper findmnt command should be `findmnt -n -o PROPAGATION /myvol2` and then you can just check `Expect(session.OutputToString()).To(Equal("shared"))` as root and for the rootless case you can use `HavePrefix("shared")`

If we care about both mountpoint then the test should check for both.

Reply:
the test is currently checking for all the mounts in the container, we could make it more precise but this way it checks for more cases (e.g. a bind mount could cause a parent mount to change its propagation)

Reply:
The thing is the cli explicitly requested a shared mount. So we need to test that this specific mount is shared otherwise the test is not checking what it should. Assuming there is a podman bug and we no longer pass shared to the runtime or there is a runtime bug which does not mount shared then this will still pass assuming there still is another unrelated mountpoint that is shared. So to me this test is not checking what it should.

Reply:
I'd still prefer the current version as it is testing more cases, but since you've a strong opinion about it, I've changed the test.  Does the new version do what you expect it to do or should I change it further?